app/src/main/java/com/keylesspalace/tusky/components/compose/view/ComposeScheduleView.kt

```diff
@@ -98,21 +100,37 @@ class ComposeScheduleView
     }
 
     fun resetSchedule() {
-        scheduleDateTime = null
+        scheduleDateTimeUtc = null
         updateScheduleUi()
     }
 
     fun openPickDateDialog() {
-        val yesterday = Calendar.getInstance().timeInMillis - 24 * 60 * 60 * 1000
+        // The earliest point in time the calendar should display. Start with current date/time
+        val earliest = calendar().apply {
+            // Add the minimum scheduling interval. This may roll the calendar over to the
+            // next day (e.g. if the current time is 23:57).
+            add(Calendar.SECOND, MINIMUM_SCHEDULED_SECONDS)
+            // Clear out the time components, so it's midnight
+            set(Calendar.HOUR_OF_DAY, 0)
+            set(Calendar.MINUTE, 0)
+            set(Calendar.SECOND, 0)
+            set(Calendar.MILLISECOND, 0)
+        }
         val calendarConstraints = CalendarConstraints.Builder()
-            .setValidator(
-                DateValidatorPointForward.from(yesterday)
-            )
+            .setValidator(DateValidatorPointForward.from(earliest.timeInMillis))
             .build()
         initializeSuggestedTime()
+
+        // Work around a misfeature in MaterialDatePicker. The `selection` is treated as
+        // millis-from-epoch, in UTC, which is good. However, it is also *displayed* in UTC
+        // instead of converting to the user's local timezone.
+        //
+        // So we have to add the TZ offset before setting it in the picker
+        val tzOffset = TimeZone.getDefault().getOffset(scheduleDateTimeUtc!!.timeInMillis)
+
         val picker = MaterialDatePicker.Builder
             .datePicker()
-            .setSelection(scheduleDateTime!!.timeInMillis)
+            .setSelection(scheduleDateTimeUtc!!.timeInMillis + tzOffset)
```

Comment:
Does it need to be subtracted then somewhere? (when taking out?)

Reply:
That conversion was already done, and happens in `onDateSet()`.

Reply:
So, it always comes out as UTC but it must be set (selection?) as UTC + time zone?

Reply:
Yes. Per the comment immediately before, it accepts the time in UTC.

What it should do is then convert that to local time before displaying it.

It doesn't. So if the UTC time is 10pm on Aug 7th, and you're in UTC+4 (so it's 2am on Aug 8th in your local timezone) the calendar showed the day as the 7th, not the 8th.